django/db/models/operators.py

```diff
@@ -0,0 +1,20 @@
+from django.db.models.expressions import CombinedExpression
+from django.db.models.fields import (
+    DateField, DateTimeField, DurationField, TimeField
+)
+
+__all__ = ()
+
+
+@CombinedExpression.dispatch(CombinedExpression.SUB, DateField, DateField)
+@CombinedExpression.dispatch(CombinedExpression.SUB, DateTimeField, DateTimeField)
+@CombinedExpression.dispatch(CombinedExpression.SUB, TimeField, TimeField)
```

Comment:
Sorry, I meant that .distpatch(CombinedExpression.SUB, DateField, DateField) will allow DateTimeField - DateField.

I guess we can just fail during execution if one of the fields is DateTimeField and the other a DateField.